src/runtime/Util/OpsHelper.cs

```diff
@@ -82,10 +82,20 @@ internal static class EnumOps<T> where T : Enum
     {
         [ForbidPythonThreads]
 #pragma warning disable IDE1006 // Naming Styles - must match Python
-        public static PyInt __int__(T value)
+        public static PyInt __int__(T value) {
```

Comment:
The `[ForbidPythonThreads]` attribute should be ensuring that when this method is called, the calling thread holds GIL.

Reply:
It's the other way round, right? If this attribute is not given, the GIL is explicitly released. If for some reason it has been released before, it will not be taken here.

Reply:
The point is: who and how releases GIL when this is called from Python?

Reply:
That's a great question, but in the end, the code here is wrong as `ForbidPythonThreads` does not take the GIL, right? After this one is merged, I'd declare the current master rc1 and start off a release-3.0 branch, ok?

Reply:
@filmor you should take this fix instead. I actually addressed ignoring `ForbidPythonThreads`: [025cc5cdcf17d33f619b8582970b8fe8b28c113e (from https://github.com/losttech/pythonnet.git branch ForbidThreadsFix)](https://github.com/losttech/pythonnet/commit/025cc5cdcf17d33f619b8582970b8fe8b28c113e)